File: 1-two-sum/two-sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        if not nums:
            return

        map = {}

        for i, num in enumerate(nums):
            complementary = target - num

            if complementary in map:
                return [map[complementary], i]
            
            map[num] = i
            
        return map
        
        

File: 1-two-sum/two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    hash = []

    for(let i = 0; i < nums.length; i++){
        comp = target - nums[i]
        if (comp in hash){
            return [i, hash[comp]];
        }
        else{
            hash[nums[i]] = i
        }
    }
};